website/www/site/content/en/documentation/io/built-in/snowflake.md

```diff
@@ -0,0 +1,738 @@
+- `.withUsernamePasswordAuth(username, password)`
+  - Sets username/password authentication.
+  - Example: `.withUsernamePasswordAuth("USERNAME", "PASSWORD")`
+- `.withOAuth(token)`
+  - Sets OAuth authentication.
+  - Example: `.withOAuth("TOKEN")`
+- `.withKeyPairAuth(username, privateKey)`
+  - Sets key pair authentication using username and [PrivateKey](https://docs.oracle.com/javase/8/docs/api/java/security/PrivateKey.html)
+  - Example: `.withKeyPairAuth("USERNAME",` [PrivateKey](https://docs.oracle.com/javase/8/docs/api/java/security/PrivateKey.html)`)`
+- `.withKeyPairPathAuth(username, privateKeyPath, privateKeyPassphrase)`
+  - Sets key pair authentication using username, path to private key file and passphrase.
+  - Example: `.withKeyPairPathAuth("USERNAME", "PATH/TO/KEY.P8", "PASSPHRASE")`
+- `.withKeyPairRawAuth(username, rawPrivateKey, privateKeyPassphrase)`
+  - Sets key pair authentication using username, private key and passphrase.
+  - Example: `.withKeyPairRawAuth("USERNAME", "PRIVATE_KEY", "PASSPHRASE")`
+
+
+**Note** - either `.withUrl(...)` or `.withServerName(...)` **is required**.
+## Pipeline options
+Use Beam’s [Pipeline options](https://beam.apache.org/releases/javadoc/2.17.0/org/apache/beam/sdk/options/PipelineOptions.html) to set options via the command line.
+### Snowflake Pipeline options
+Snowflake IO library supports following options that can be passed via the [command line](https://beam.apache.org/documentation/io/built-in/snowflake/#running-main-command-with-pipeline-options) by default when a Pipeline uses them:
+
+`--url` Snowflake's JDBC-like url including account name and region without any parameters.
+
+`--serverName` Full server name with account, zone and domain.
+
+`--username` Required for username/password and Private Key authentication.
+
+`--oauthToken` Required for OAuth authentication only.
+
+`--password` Required for username/password authentication only.
+
+`--privateKeyPath` Path to Private Key file. Required for Private Key authentication only.
+
+`--rawPrivateKey` Private Key. Required for Private Key authentication only.
+
+`--privateKeyPassphrase` Private Key's passphrase. Required for Private Key authentication only.
+
+`--stagingBucketName` External bucket path ending with `/`. I.e. `gs://bucket/`. Sub-directories are allowed.
+
+`--storageIntegrationName` Storage integration name
+
+`--warehouse` Warehouse to use. Optional.
+
+`--database` Database name to connect to. Optional.
+
+`--schema` Schema to use. Optional.
+
+`--table` Table to use. Optional.
+
+`--query` Query to use. Optional.
+
+`--role` Role to use. Optional.
+
+`--authenticator` Authenticator to use. Optional.
+
+`--portNumber` Port number. Optional.
+
+`--loginTimeout` Login timeout. Optional.
+
+`--snowPipe` SnowPipe name. Optional.
+
+In the connector code SnowflakePipelineOptions looks following (and it can be [extended](https://beam.apache.org/documentation/io/built-in/snowflake/#extending-pipeline-options)):
```

Comment:
I would remove whole this section with pipeline options. It's not relevant for Beam users

Reply:
good idea IMO 👍

Reply:
Removed